tests/integration/records/single_db/test_records_unload_datetime.py

```diff
@@ -0,0 +1,376 @@
+            if uses_pandas:
+                # We're going to need to be sure to use hints that
+                # work in Pandas
+                addl_hints.update(pandas_compatible_addl_hints)
+                if 'AM' in datetimeformattz:
+                    # Pandas is more limited in hint support than it
+                    # should be:
+                    #
+                    # https://github.com/bluelabsio/records-mover/issues/143
+                    expect_am_failure = True
+
+            records_format = RecordsFormat(variant=VARIANT_FOR_DB[self.engine.name],
+                                           hints={
+                                               'datetimeformattz': datetimeformattz,
+                                               'compression': None,
+                                               'header-row': False,
+                                               **addl_hints,  # type: ignore
+                                           })
+            expect_pandas_failure = not self.has_pandas() and uses_pandas
+            try:
+                csv_text = self.unload(column_name='timestamptz',
+                                       records_format=records_format)
+                self.assertIn(csv_text, [create_sample(datetimeformattz) + "\n",
+                                         create_sample(datetimeformattz).
+                                         replace('-00', '+00') + "\n",
+                                         create_sample(datetimeformattz).replace('-00', '') +
+                                         ".000000\n",
+                                         # TODO: Should this be necessary?
+                                         create_sample(datetimeformattz) + ".000000\n",
+                                         # TODO: Should this be necessary?
+                                         create_sample(datetimeformattz).replace('-00', '') +
+                                         '.000000+0000\n',
+                                         # TODO: Should this be necessary?
+                                         create_sample(datetimeformattz) +
+                                         f":{SAMPLE_SECOND:02d}.000000\n"
+                                         ],
+                              f"from datetimeformattz {datetimeformattz} and "
+                              f"addl_hints {addl_hints}")
+            except ModuleNotFoundError as e:
+                if 'pandas' in str(e) and expect_pandas_failure:
+                    # as expected
+                    continue
+                else:
+                    raise
+            except NotImplementedError as e:
+                if expect_am_failure and 'AM' in str(e):
+                    # as expected
+                    continue
+                else:
+                    raise
+            self.assertFalse(expect_pandas_failure)
+            self.assertFalse(expect_am_failure)
+
+    def test_unload_timeonly(self) -> None:
```

Comment:
C901 'RecordsUnloadDatetimeIntegrationTest.test_unload_timeonly' is too complex (17)